src/Main/ViewModeButtons.jsx

```diff
@@ -1,136 +1,78 @@
-          }}
-        >
-          <Icon className="viewModeButton">
-            <img
-              style={{ height: '24px', width: '24px' }}
-              src={yellowPlaneIconDataUri}
-            />
-          </Icon>
-        </ToggleButton>
-      </Tooltip>
-      <Tooltip
-        ref={zPlaneButton}
-        title="z plane [3]"
-        PopperProps={{
-          anchorEl: zPlaneButton.current,
-          disablePortal: true,
-          keepMounted: true
+    <OverlayTrigger
+      transition={false}
+      overlay={<Tooltip> {props.title} </Tooltip>}
+    >
+      <Button
+        className={cn('icon-button', {
+          checked: state.context.main[enableButton]
+        })}
+        onClick={() => {
+          send({ type: 'VIEW_MODE_CHANGED', data: props.dataName })
         }}
+        variant="secondary"
       >
-        <ToggleButton
-          size="small"
-          className="toggleButton"
-          value="zplane"
-          selected={state.context.main.zPlaneButton}
-          onClick={() => {
-            send({ type: 'VIEW_MODE_CHANGED', data: 'ZPlane' })
-          }}
-        >
-          <Icon className="viewModeButton">
-            <img
-              style={{ height: '24px', width: '24px' }}
-              src={greenPlaneIconDataUri}
-            />
-          </Icon>
-        </ToggleButton>
-      </Tooltip>
-      <Tooltip
-        ref={volumeButton}
-        title="Volume [4]"
-        PopperProps={{
-          anchorEl: volumeButton.current,
-          disablePortal: true,
-          keepMounted: true
-        }}
-      >
-        <ToggleButton
-          size="small"
-          className="toggleButton"
-          value="volume"
-          selected={state.context.main.volumeButton}
-          onClick={() => {
-            send({ type: 'VIEW_MODE_CHANGED', data: 'Volume' })
-          }}
-        >
-          <Icon className="viewModeButton">
-            <img
-              style={{ height: '24px', width: '24px' }}
-              src={volumeIconDataUri}
-            />
-          </Icon>
-        </ToggleButton>
-      </Tooltip>
-    </div>
+        <Image src={props.imageSrc}></Image>
+      </Button>
+    </OverlayTrigger>
   )
 }
 
+function ViewModeButtons(props) {
+  const { service } = props
+
+  const buttonList = [
+    {
+      title: 'X plane [1]',
+      dataName: 'XPlane',
+      imageSrc: redPlaneIconDataUri
+    },
+    {
+      title: 'Y plane [2]',
+      dataName: 'YPlane',
+      imageSrc: yellowPlaneIconDataUri
+    },
+    {
+      title: 'Z plane [3]',
+      dataName: 'ZPlane',
+      imageSrc: greenPlaneIconDataUri
+    },
+    {
+      title: 'Volume [4]',
+      dataName: 'Volume',
+      imageSrc: volumeIconDataUri
+    }
+  ]
+  const listItems = buttonList.map(({ title, dataName, imageSrc }) => (
+    <ViewButton
+      key={title}
+      title={title}
+      dataName={dataName}
+      imageSrc={imageSrc}
+      service={service}
+    ></ViewButton>
+  ))
+  return listItems
```

Comment:
A masterpiece!  

My style would be to skip the variable names, like so: 
```
return [ { stuff }, { stuff } ].map((stuff ) => (<Component stuff>))
```

But you do you.

Reply:
Wow, thanks Paul!